Docker/Dockerfile

```diff
@@ -0,0 +1,47 @@
+# starting from a base image of Ubuntu
+FROM ubuntu:latest
+
+# **** Base System Initialization and Setup  ****
+
+# Install the required libraries:
+# - Gcc for compiler
+# - Git for source code access
+# - Cmake for build configuration
+# - Pkg_config to aide CMake in package sourcing see: https://autotools.io/pkgconfig/pkg_check_modules.html
+# - libssl for access to build requirements for encryption protocols
+RUN apt-get update \
+	&& apt-get upgrade -y \
+	&& apt-get install -y gcc git cmake pkg-config libssl-dev 
```

Comment:
I'm not sure if explicitly requiring `gcc` is a good idea. If Debian switches to `clang` one day, I don't want to hold them back. Maybe we should use the `build-essentials` package instead?